### ISSUE TITLE
BUG: broken datetime when using to_sql()

### ISSUE THREAD

**Issue by teweihuang** [pandas-dev/pandas#55005]
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
import pandas as pd
import sqlite3

today = pd.to_datetime("2023/9/5")
ele = [1, 2, 3]
df = pd.DataFrame({'date': [today] * len(ele), 'data': ele})
df.set_index('date', inplace=True)

conn = sqlite3.connect('date.db')
df.to_sql('test_date', conn, if_exists='replace')
conn.close()
```

### Issue Description

the date in db becoming like this : 1693843200

<img width="158" alt="screenshot1" src="https://github.com/pandas-dev/pandas/assets/50105867/21696ba1-be5f-4c51-8466-57a512afcc83">
<img width="333" alt="screenshot2" src="https://github.com/pandas-dev/pandas/assets/50105867/9d7b417c-3f9d-489b-927c-29b3fd431df6">

the same code was working properly in 2.0.3, the date was correctly identified into YYYY-mm-dd.

I understand it's suggested to use utc=True, but please don't just make db broken instantly, that hurts.

it would be appreciated if devs could take a look.

### Expected Behavior

It should be identified as YYYY-mm-dd, in this case, 2023-09-05 00:00:00, when using .to_sql() saving in db.

### Installed Versions

<details>
INSTALLED VERSIONS
------------------
commit              : ba1cccd19da778f0c3a7d6a885685da16a072870
python              : 3.11.5.final.0
python-bits         : 64
OS                  : Windows
OS-release          : 10
Version             : 10.0.22621
machine             : AMD64
processor           : AMD64 Family 23 Model 113 Stepping 0, AuthenticAMD
byteorder           : little
LC_ALL              : None
LANG                : None
LOCALE              : Chinese (Traditional)_Taiwan.950
pandas              : 2.1.0
numpy               : 1.25.2
pytz                : 2023.3
dateutil            : 2.8.2
setuptools          : 68.1.2
pip                 : 23.2.1
Cython              : None
pytest              : None
hypothesis          : None
sphinx              : None
blosc               : None
feather             : None
xlsxwriter          : None
lxml.etree          : 4.9.3
html5lib            : None
pymysql             : None
psycopg2            : None
jinja2              : None
IPython             : None
pandas_datareader   : None
bs4                 : 4.12.2
bottleneck          : None
dataframe-api-compat: None
fastparquet         : None
fsspec              : None
gcsfs               : None
matplotlib          : 3.7.2
numba               : None
numexpr             : None
odfpy               : None
openpyxl            : None
pandas_gbq          : None
pyarrow             : None
pyreadstat          : None
pyxlsb              : None
s3fs                : None
scipy               : 1.11.2
sqlalchemy          : None
tables              : None
tabulate            : None
xarray              : None
xlrd                : None
zstandard           : None
tzdata              : 2023.3
qtpy                : None
pyqt5               : None

</details>

**Comment by asishm** [comment]
Thanks for the report. Duplicate of #54877

**Comment by mroeschke** [comment]
Thanks for the report. This will be fixed in 2.1.1
